packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md

```diff
@@ -0,0 +1,36 @@
+Check for an existing session in two steps.
+
+1. Initialize the authorization state manager (`authStateManager`) from a saved session.
+2. Verify that the access token is still valid by checking the expiry.
+
+The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+
+Update the `tokenExpired` utility function to check for token expiry:
+
+```swift
+func tokenExpired(_ tokenString: String?) -> Bool {
+  guard let accessToken = tokenString,
+  let tokenInfo = OKTIDToken.init(idTokenString: accessToken) else {
+    return false
+  }
+    
+  return Date() > tokenInfo.expiresAt
+}
+```
+
+Then add this code to the end of `configureSDK` to initialize the state manager from a saved session and to check if the token is still valid:
```

Comment:
I think the following is a bit clearer and more succinct:

Then initialize the state manager from a saved session if the token is valid by adding this code to the end of `configureSDK`.

Reply:
Agreed. Updated.